src/main/java/org/cp/confisual/nevisauth/Entry.java

```diff
@@ -1,16 +1,42 @@
 package org.cp.confisual.nevisauth;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Stack;
+
 import lombok.Getter;
 import lombok.RequiredArgsConstructor;
 import lombok.Setter;
+import org.cp.confisual.plantuml.NevisAuthObject;
+import org.cp.confisual.plantuml.NevisAuthVisitor;
 
 @RequiredArgsConstructor
 @Getter
-public class Entry {
+public class Entry implements NevisAuthObject {
 
   private final String method;
   private final String authStateName;
   private final String selector;
   @Setter
   private AuthState authState;
+  public static List<String> existingAuthStates = new ArrayList<>();
+  @Override
+  public void accept(NevisAuthVisitor nevisAuthVisitor) {
+    nevisAuthVisitor.visit(this);
+
+    Stack<AuthState> authStateS = new Stack<>();
```

Comment:
Just visit the referenced authstate of each entry here, the logic how to go further to be implemented in each authstate.